yarn-project/noir-contracts/src/contracts/non_native_token_contract/src/transparent_note.nr

```diff
@@ -1,75 +1,147 @@
-    secretHash: Field,
+    secret_hash: Field,
+    // the fields below are not serialised/deserialised
+    secret: Field,
+    leafIndex: Field,
+    header: NoteHeader,
 }
 
 impl TransparentNote {
-    fn new(amount: Field, secretHash: Field) -> Self {
-        Self { amount, secretHash }
+
+    // CONSTRUCTORS
+
+    fn new(amount: Field, secret_hash: Field) -> Self {
+        TransparentNote {
+            amount: amount,
+            secret_hash: secret_hash,
+            secret: 0,
+            leafIndex: 0,
+            header: NoteHeader::empty(),
+        }
     }
 
+    // new oracle call primitive
+    // get me the secret corresponding to this hash
     fn new_from_secret(amount: Field, secret: Field) -> Self {
-        let secretHash = TransparentNote::compute_secret_hash(secret);
         TransparentNote {
-            amount,
-            secretHash,
+            amount: amount,
+            secret_hash: TransparentNote::compute_secret_hash(secret),
+            secret: secret,
+            leafIndex: 0,
+            header: NoteHeader::empty(),
         }
     }
 
-    // Gets the value of the commitment
-    // TODO(maddiaa): this will need to be hashed with a slot to keep it unique, which slot to pick? https://github.com/AztecProtocol/aztec-packages/issues/847 
-    fn get_commitment(self: Self) -> Field {
-        pedersen([self.amount, self.secretHash])[0]
+
+    // STANDARD NOTE_INTERFACE FUNCTIONS
+
+    fn serialise(self) -> [Field; TRANSPARENT_NOTE_LEN] {
+        [self.amount, self.secret_hash]
     }
 
-    fn consume_in_secret(self: Self, context: &mut Context, root: Field, secret: Field) {
-        // Get the commitment value (before silo)
-        let innerCommitment = self.get_commitment();
+    fn deserialise(preimage: [Field; TRANSPARENT_NOTE_LEN]) -> Self {
+        TransparentNote {
+            amount: preimage[0],
+            secret_hash: preimage[1],
+            secret: 0,
+            leafIndex: 0,
+            header: NoteHeader::empty(),
+        }
+    }
 
-        // Get the commitment data, (where it is in the db)
-        let commitment_oracle_call = get_commitment(innerCommitment);
-        let commitment_data = make_commitment_getter_data(commitment_oracle_call, 0);
-        let siloed_commitment = commitment_data.message; // oracle call returns siloed commitment
+    fn compute_note_hash(self) -> Field {
+        dep::std::hash::pedersen([
+            self.amount,
+            self.secret_hash,
+        ])[0]
+    }
 
-        // Let the kernel perform the read.
-        context.push_read_request(siloed_commitment);
+    fn compute_nullifier(self) -> Field {
+        // TODO(https://github.com/AztecProtocol/aztec-packages/issues/1386): should use
+        // `compute_note_hash_for_read_or_nullify` once public functions inject nonce!
+        let siloed_note_hash = compute_siloed_note_hash(TransparentNoteMethods, self);
+        pedersen([self.secret, siloed_note_hash, self.leafIndex])[0]
```

Comment:
in this construction will leaf index ever not be 0?

Reply:
Good catch. Realistically we should just get rid of this leafIndex. Once we have nonces all working for public, the commitments themselves will be unique and this won't be necessary. I'll just remove leafIndex here for now and will link the ticket to add nonces here.